src/components/web3/SendCryptoPopup.tsx

```diff
@@ -0,0 +1,476 @@
+import { Text } from "../Text";
+import { css } from "@emotion/react";
+import { useEffect, useRef, useState } from "react";
+import { generateSIWEForCrypto } from "./api";
+import { IJitsiMeetApi } from "../../jitsi/types";
+import { getNonce } from "./api";
+import { SiweMessage } from "siwe";
+import { verifyMessage, parseUnits, Transaction } from "ethers";
+import { Buffer } from "buffer";
+import { AllowedERC20Tokens, sendCrypto } from "./send-crypto";
+const popupBaseCSS = css`
+  position: absolute;
+  margin: 10px;
+  width: 30%;
+  height: 10%;
+  border-radius: 0.5rem;
+  background-color: rgb(128, 128, 128);
+  z-index: 99999;
+  font-family: system-ui;
+`;
+
+const popupCSSTopLeft = css`
+  top: 0;
+  left: 0;
+`;
+
+const popupCSSBottomRight = css`
+  bottom: 0;
+  right: 0;
+`;
+
+const popupCSSCetner = css`
+  top: 50%;
+  left: 50%;
+  -webkit-transform: translate(-50%, -50%);
+  transform: translate(-50%, -50%);
+`;
+
+const buttonCSS = css`
+  margin: 10px;
+  width: 20%;
+  cursor: pointer;
+  border-radius: 0.2rem;
+`;
+
+export interface CryptoTransactionParams {
+  senderDisplayName: string;
+  sender: string;
+  recipient: string;
+  recipientDisplayName: string;
+  amount: number;
+  token: AllowedERC20Tokens;
+  nonce: string;
+}
+
+export interface SIWEReturnParams {
+  proof: {
+    signer: string;
+    signature: string;
+    payload: string;
+  };
+  method: string;
+}
+
+interface TransactionPendingResolution {
+  senderDisplayName: string;
+  siwe: SIWEReturnParams;
+}
+
+interface CryptoSendMessage {
+  type: "crypto";
+  msgType: "REQ" | "REJECT" | "SIGNED";
+  payload: SIWEReturnParams | CryptoTransactionParams | string;
+}
+
+interface CryptoPopupProps {
+  incomingRequests: CryptoTransactionParams[];
+  setIncomingRequests: React.Dispatch<
+    React.SetStateAction<CryptoTransactionParams[]>
+  >;
+  web3Address: string;
+  jitsi: IJitsiMeetApi;
+}
+
+export const CryptoRecievePopup: React.FC<CryptoPopupProps> = ({
```

Comment:
This is hard to follow.  I would decompose this into constituent components `CryptoRecievePopup`, `CryptoSendPopup`, `CryptoOOBPopup`, `CryptoWrapper`.

Reply:
This change has been made in 651df37ff44c8c4e31cd9653c98e6e79bd4412d7. The four components are their own files, and also definitions/css/components that are common to them are in `common.ts`, `common-css.ts`, and `common-components.tsx` resp. These are all in a new folder `web3/crypto`.